server/anoixo-server/text_providers/Nestle1904LowfatProvider.py

```diff
@@ -238,6 +238,14 @@ def _build_query_string(self, query: TextQuery) -> str:
         # else -1
         matched_word_query_switch = '\nelse '.join(word_query_index_getters) + '\nelse -1'
 
+        # Goes through allowed_attributes and makes a string to include in the XQuery that gets all their data
+        # Each line will look something like:
+        # "class": data($w/@class),
+        generated_attributes = ""
+        for attribute in allowed_attributes:
+            generated_attributes = generated_attributes + """,
```

Comment:
Thanks for adding that comment! A couple small things:
- Can you use format strings instead of doing string concatenation? For instance, `f'data($w/@{attribute})', instead of `"data($w/@" + attribute + ")"`
- Throughout the rest of this function, when I need to loop to generate multiple pieces of code and then concatenate them, I usually make an array with each piece (for instance, `['"lemma": data($w/@lemma)', '"normalized": data($w/@normalized)', ...]`) then use `.join` to connect them up (in this case, you'd probably do `',\n'.join(attribute_selectors)` or something like that). There's an example on lines 227-228. Can you follow that pattern instead of doing this loop?

Reply:
I chose to use format strings and arrays + .join in the rest of this function since I found them to be easier to read and understand (and thus less error-prone) than the alternatives. (For instance, it took me a minute to realize when I looked at your loop that the resulting string would start with a comma + newline, as well as putting a comma + newline between each generated attribute selector.) So that's part of why I'm asking you to use them, but it's also just generally a good idea to use a consistent style/approach, especially for a function like this that's already gnarly enough :worried: